src/commands/cargo.ts

```diff
@@ -1,6 +1,23 @@
 import * as io from '@actions/io';
 import * as core from '@actions/core';
 import * as exec from '@actions/exec';
+import * as cache from '@actions/cache';
+import * as http from '@actions/http-client';
+import * as path from 'path';
+
+export async function resolveVersion(crate: string): Promise<string> {
+    const url = `https://crates.io/api/v1/crates/${crate}`;
+    const client = new http.HttpClient(
+        '@actions-rs (https://github.com/actions-rs/)',
+    );
+
+    const resp: any = await client.getJson(url);
+    if (resp.result == null) {
+        throw new Error('Unable to fetch latest crate version');
+    }
+
+    return resp.result['crate']['newest_version'];
+}
```

Comment:
This function was moved from actions-rs/install, since it is required in this repo to resolve the crate version, when `latest` is chosen. Otherwise it would not be possible to update the cache, since the name `<crate>-latest-<key>` would always be the same